examples/upload.js

```diff
@@ -44,16 +43,33 @@ async function run() {
   }
 
   try {
-    for (let i = 0; i < attempts; i++) {
-      await Promise.all([
-        permanent.record.uploadFromUrl({displayName: 'image', uploadUri: imageUrl, uploadFileName: imageName}, folderForSet),
-      ]);
-    }
-    logTime();
-    console.log('all success!');
+      let record = {};
+      // note: this is the size of our example photo
+      record.size = 98894;
+      const fileType = 'image/jpeg';
+      // call getappfolders here, get parentfolder_linkid
+      const appFolders = await permanent.folder.getAppFolders();
+      const etherpadFolder = await permanent.folder.create(`Etherpad`, appFolders[0]);
+      // we expect to get size and type from the client
+      // assemble into a recordvo
+      record.parentFolder_linkId = etherpadFolder.parentFolder_linkId;
+      const testPresigned = await permanent.record.getPresignedUrl(fileType, record);
+      // return presigned info to the client so it can post directly to the destination
+      // then call registerRecordAndAddStorage
+      const archiveId = permanent.getArchiveId();
+      record.parentFolderId = etherpadFolder.parentFolderId;
+      record.parentFolder_linkId = etherpadFolder.parentFolder_linkId;
+      record.displayName = 'puppy';
+      record.uploadFileName = imageName;
+      record.derivedCreatedDT = '2021-04-08';
+      record.archiveId = archiveId;
+      permanent.record.registerRecord(record, testPresigned.value.destinationUrl);
```

Comment:
I think this needs to be updated to match:

```suggestion
      permanent.record.registerRecordAndAddStorage(record, testPresigned.value.destinationUrl);
```

Reply:
Ah, yes, I haven't really kept the example script up to date because I've been changing it around so much, but I can get the latest version committed now.

Reply:
I have now committed this, though I'm not sure how useful it is to anyone but me.